include/device.h

```diff
@@ -553,10 +580,11 @@ static inline int z_impl_device_usable_check(const struct device *dev)
  * Indicates whether the provided device pointer is for a device known to be
  * in a state where it can be used with its standard API.
  *
- * This can be used with device pointers captured from DEVICE_DT_GET(), which
- * does not include the readiness checks of device_get_binding().  At minimum
- * this means that the device has been successfully initialized, but it may
- * take on further conditions (e.g. is not powered down).
+ * This can be used with device pointers captured from DEVICE_DT_GET() or
+ * device_from_name(), which do not include the readiness checks of
+ * device_get_binding().  At minimum this means that the device has been
+ * successfully initialized, but it may take on further conditions (e.g. is
```

Comment:
This paragraph is starting to feel like it's doing too many things. The antecedent for "this" in "this means" was unclear to me after the discussion of all the different APIs that "this" function (which is the antecedent in "this can be used" earlier) is related to. How about splitting it up:

```suggestion
 * device_get_binding(). 
 * 
 * At minimum, "ready" means that the device has been successfully initialized,
 * but it may take on further conditions (e.g. is
```

Reply:
done, thanks.

Reply:
Replacing device_get_binding() with a function returning an error code (like Linux match() ) would be useful, so can distinguish "not ready" and "doesn't exist". Otherwise unclear what a good idiom is for a caller to wait for a device to initialize. Encountered this (v2.4) where main tread early on calls device_get_binding() for a board LED that is driven by PWM driver that wasn't yet fully initialized - got intermittent board startup failures. Alternatively, could deprecate this function, forcing to always use DEVICE_DT_GET and ensure all driver api calls return an -EREADY or similar code so caller can retry. Deciding how long to wait e.g for a sensor that is dependent on other drivers is also tricky.